internal/format/templates/asciidoc_document.tmpl

```diff
@@ -48,18 +48,27 @@
     {{ end }}
 {{ end -}}
 
-{{- if .Settings.ShowResources -}}
+{{- if or .Settings.ShowResources .Settings.ShowDataSources -}}
     {{ indent 0 "=" }} Resources
     {{ if not .Module.Resources }}
         No resources.
     {{ else }}
         The following resources are used by this module:
         {{ range .Module.Resources }}
-            {{ if eq (len .URL) 0 }}
-            - {{ .Spec }} {{ printf "(%s)" .GetMode -}}
-            {{- else -}}
-            - {{ .URL }}[{{ .Spec }}] {{ printf "(%s)" .GetMode -}}
-            {{- end }}
+            {{- if $.Settings.ShowResources -}}{{- if eq "resource" (printf "%s" .GetMode) }}
```

Comment:
For simplicity and not duplicating code these two if blocks can be combined to one. Something like:

```
{{- $isResource := and $.Settings.ShowResources ( eq "resource" (printf "%s" .GetMode)) }}
{{- $isDataResource := and $.Settings.ShowDataSources ( eq "data source" (printf "%s" .GetMode)) }}
{{- if or $isResource $isDataResource }}
    {{ if eq (len .URL) 0 }}
     .....
```

Reply:
Beautiful solution! I was with doubt about my solution, this your suggestion is wonderful!